src/main/java/org/jenkinsci/plugins/github_branch_source/ExcludeArchivedTrait.java

```diff
@@ -0,0 +1,54 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import hudson.Extension;
+import jenkins.scm.api.trait.SCMNavigatorContext;
+import jenkins.scm.api.trait.SCMNavigatorTrait;
+import jenkins.scm.api.trait.SCMNavigatorTraitDescriptor;
+import jenkins.scm.impl.trait.Selection;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A {@link Selection} trait that will restrict the discovery of repositories that have been archived.
+ */
+public class ExcludeArchivedTrait extends SCMNavigatorTrait {
```

Comment:
Repositories are the only thing that gets archived on GitHub - for now at least.  Unless you feel strongly about it, I'd rather have a longer unambiguous class name.   

This will result in changes throughout the rest of the PR. 
```suggestion
public class ExcludeArchivedRepositoriesTrait extends SCMNavigatorTrait {
```

Reply:
Renamed.